### ISSUE TITLE
Frontend testing 23. 11. 2019

### ISSUE THREAD

**Issue by jbrezmorf** [GeoMop/visip#28]
#Notes from testing:

1. Error, when sterting from different directory:

    {TB_Evaluation} ~/workspace/visip$ python3 src/frontend/VISIP.py 
    Traceback (most recent call last):
      File "src/frontend/VISIP.py", line 13, in <module>
      os.chdir(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
      FileNotFoundError: [Errno 2] No such file or directory: '/home/jb/workspace/src'

2. How important is the dependency on 'pyqtgraph'? 
3. Detach connection Input/Output: doesn't work, Left click always initialize new connection.
        Alt + Left on Linux initializes move of the window. 
3. Can not rename the action call in its icon. Can edit the name in Parameters editor, but the scene is not updated. 
4. Error when connectin more actions to the result inputs:
   ```
   File "/home/jb/workspace/visip/src/frontend/widgets/scene.py", line 233, in update_unconected
    self.unconnected_actions.pop(action.name, None)
   AttributeError: 'NoneType' object has no attribute 'name'
   One can connect output of the result action to the input of other action.
   ````
5. Error on right click to the scene (only after moving the action block??):
```
  File "/home/jb/workspace/visip/src/frontend/widgets/scene.py", line 165, in add_action
    assert False, "Action isn't supported by scene!"
AssertionError: Action isn't supported by scene!
```
... seems that the problem is the first item of the contet menu "New action" this triggers the error.

6. The output of the result action can be connected to the input. It should be similar to the input of Slots.

7. Shift + wheel - big step vertical pan. Should rather pan horizontally.

8. allow selection of actions by a rectangle

9. Support Ctrl+C, Ctrl+X, Ctrl+V between workflows.

10. Toolbox: missing function to open the module (in new tab).

11. Better interaction with the Module panel:
   Purpose of list of action calls in the panel? Direct operations without context menu. Probably need to display more workflows together (window splitting or in single scene ??)

12. Parameters Editor: 
   - editing the action call name hides the parameters.
   - can not edit non-constant parameters

13. A connection to the constant parameters is allowed. This breaks module export.
    Probablly needs special treatment in the backend as well. How to report errors in connections?

**Comment by jbrezmorf** [comment]
List of errors and new ideas. We will discuss it personally, possibly creating new issues.

**Comment by jbrezmorf** [comment]
ad 3. change from Alt+left to Shift+left
ad 4. make separate issue, need support in backend
ad 7. to be implemented, Shift+Weel - horizontal pan
ad. 9. new issue, need data serialization??, need not to use external clipboard
ad. 10. new issue
ad. 11. Show workflow on doble-click. 
ad 11. Zalozky pro moduly, podtim zalozky pro workflow  (a dataclass). Toto samostatne pro pripadnych vice oken (split). 
12.  - todo: zakazat zabalovani pri dvojkliku
      - new issue pro editaci nekonstant, textova editace propojeni konvertoru
13. new issue, fixing coloring and reporting errors with connections

**Comment by TomasBlazek1** [comment]
All bugs and sugestions should be resolved or moved to independent issue.
